src/core/annotation.js

```diff
@@ -1610,11 +1610,14 @@ class ChoiceWidgetAnnotation extends WidgetAnnotation {
       }
     }
 
-    // Determine the field value. In this case, it may be a string or an
-    // array of strings. For convenience in the display layer, convert the
-    // string to an array of one string as well.
-    if (!Array.isArray(this.data.fieldValue)) {
+    // The field value can be `null` if no item is selected, a string if one
+    // item is selected or an array of strings if multiple items are selected.
+    // For consistency in the API and convenience in the display layer, we
+    // always make the field value an array with zero, one or multiple items.
+    if (isString(this.data.fieldValue)) {
       this.data.fieldValue = [this.data.fieldValue];
+    } else if (!this.data.fieldValue) {
+      this.data.fieldValue = [];
     }
```

Comment:
Furthermore, in the Array-case: Since the comment says that it should be an array of strings, don't you also need to iterate through the array and validate/use `stringToPDFString` on all elements?

Reply:
This patch is only about handling the possible `null` value that's in the spec. It doesn't fix #12233, but is only something I noticed while debugging that. Fixing #12233 would most likely require using `stringToPDFString`, but I've got a way to do that in a more centralized manner. That's however part of the patch that will address that issue.

(Same for the comment above.)

Reply:
Since I couldn't help but noticing the lack of `stringToPDFString` in this this code, I had to ask.
Assuming that this will all be fixed soon anyway, this seems fine to me as-is.